src/loadConfigurationFile.js

```diff
@@ -60,5 +60,12 @@ module.exports = function(configPath) {
             throw new Error('Could not load required module loading for ' + chalk.underline(configPath));
         }
     }
-    return require(configPath);
-}
\ No newline at end of file
+    var config = require(configPath);
+    if (typeof config === 'function')
+        return config(require('minimist')(process.argv, { '--': true }).env || {});
```

Comment:
took me a bit to remember that the es module export can be an object too ^^
not sure if you like the way i split the methods